internal/csi-common/driver.go

```diff
@@ -33,7 +33,10 @@ type CSIDriver struct {
 	nodeID  string
 	version string
 	// topology constraints that this nodeserver will advertise
-	topology     map[string]string
+	topology map[string]string
+	// crushLocationMap that will be used to enable read affinity.
+	CrushLocationMap map[string]string
```

Comment:
> does this need to be exported? Ideally it isn't.

Yes, it needs to exported to be accessible to individual drivers to use in rpc function call.

Reply:
Why is then this different from `topology`?

Reply:
> Why is then this different from `topology`?

Currently processing required for topology is identical to cephfs and rbd (topology feature is itself in beta stage with lot of TODOs), 
Whereas processing required for CrushLocation maybe different for cephfs and rbd,(only rbd support read affinity at the moment.). Therefore, its better to let individual drivers handle it.

Reply:
The `CSIDriver` struct does not expose any attributes at the moment. It would be better to provide access/get function(s) for it instead. Sortof what you have in `internal/util/crushlocation.go`, but maybe even further and generate the `crush_location=` rbd commandline option?